src/components/TopicsBySituation/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { useState } from "react";
+import "swiper/css";
+import { Swiper, SwiperSlide } from "swiper/react";
+import { EffectCards } from "swiper/modules";
+import type { Swiper as SwiperType } from "swiper";
+
+import * as S from "./styled";
+import Header from "../Header";
+import { topics } from "./data";
+
+const TOPICS_LENGTH_TO_SHOW = 3;
+
+const SWIPER_CONFIG = {
+  slidesPerView: 1,
+  effect: "cards" as const,
+  modules: [EffectCards],
+  className: "stacked-card-swiper",
+  style: { height: "29.86rem" },
+  cardsEffect: {
+    perSlideOffset: 0,
+    perSlideRotate: 0,
+    rotate: false,
+    slideShadows: false,
+  },
+  direction: "horizontal" as const,
+  loop: false,
+};
+
+const TopicsBySituation = () => {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [isLastSlide, setIsLastSlide] = useState(false);
+  // 마지막 슬라이드를 위해 빈 문자열을 추가
+  const slides = [...topics, ""];
+
+  const topicsToShow = topics.slice(
+    currentIndex,
+    Math.min(currentIndex + TOPICS_LENGTH_TO_SHOW, topics.length),
+  );
+
+  const handleSlideChange = (swiper: SwiperType) => {
+    setCurrentIndex(Math.min(swiper.activeIndex, topics.length - 1));
+    setIsLastSlide(swiper.isEnd);
+  };
+
+  return (
+    <>
+      <Header title="소개팅 토픽" />
+      <S.Main>
+        <S.SituationBox>
+          <S.Situation>#소개팅</S.Situation>
+          <S.ViewAllTopicsButton>전체 토픽 둘러보기 &gt;</S.ViewAllTopicsButton>
+        </S.SituationBox>
+
+        <S.CardStackContainer>
+          {isLastSlide ? (
+            <div>모든 토픽을 확인하셨습니다!</div>
+          ) : (
+            <Swiper
+              {...SWIPER_CONFIG}
+              onSlideChange={handleSlideChange}
+              slidesPerView={1}
+            >
+              {slides.map((_, index) => (
+                <SwiperSlide key={`slide-${index}`}>
+                  <S.CardStack>
+                    {topicsToShow.map((stackTopic, stackIndex) => (
+                      <S.StackedCard
+                        key={`${stackTopic}-${stackIndex}`}
+                        order={(stackIndex + 1) as 1 | 2 | 3}
+                      >
+                        {stackTopic}
+                      </S.StackedCard>
+                    ))}
+                  </S.CardStack>
+                </SwiperSlide>
+              ))}
+            </Swiper>
+          )}
+        </S.CardStackContainer>
+
+        {!isLastSlide ? (
+          <S.ProgressContainer>
+            <S.ProgressText>
+              {currentIndex + 1}/{topics.length}
+            </S.ProgressText>
+          </S.ProgressContainer>
+        ) : null}
```

Comment:
```suggestion
        {!isLastSlide && (
          <S.ProgressContainer>
            <S.ProgressText>
              {currentIndex + 1}/{topics.length}
            </S.ProgressText>
          </S.ProgressContainer>)}
```
&& 연산자로도 해결할 수 있을 것 같아요~

Reply:
@wo-o29 &&로 사용하면 false일 때 0이 렌더링되는 케이스가 있어서 이렇게 했어요

Reply:
0은 리액트에서 무시되는 값(null, undefined, boolean)이 아니라서 사용할 때 `isLastSlide.length !== 0` 이런식으로 Boolan 값이 나오게 사용하면 될 것 같아요!
https://velog.io/@woogur29/React%EC%97%90%EC%84%9C-%EB%AC%B4%EC%8B%9C%EB%90%98%EB%8A%94-%EA%B0%92null-undefined-boolean%EB%93%A4%EC%9D%84-%EC%96%B4%EB%96%BB%EA%B2%8C-%EC%B2%98%EB%A6%AC%ED%95%A0%EA%B9%8C#%EB%AC%B4%EC%8B%9C%EB%90%98%EB%8A%94-%EA%B0%92-%ED%99%9C%EC%9A%A9%ED%95%98%EA%B8%B0%EC%A1%B0%EA%B1%B4%EB%B6%80-%EB%A0%8C%EB%8D%94%EB%A7%81